[13.0] [IMP] l10n_ro - Improve Romanian Localisation

feketemihai opened the pull request:
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:




--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

feketemihai commented:
@jcr-odoo I created a new PR, https://github.com/odoo/odoo/pull/39577, so we can close the other 2:
https://github.com/odoo/odoo/pull/22105 and https://github.com/odoo/odoo/pull/23704.

feketemihai commented:
@jcr-odoo I still have issue with Not Deductible VAT Taxes, because of the requirements needed from Romanian Accounting procedures, because there is should split the base in 2, and for Colected VAT one it should only be 50% (ussualy) from the Deducted VAT base, and i cannot split it because of the restriction to have only one "base" distribution in invoice_repartition_line_ids and refund_repartition_line_ids, and even setting a different percentage, it always get 100% of the base. I don't know if you can understand, i already mentioned this issue in the other PR, https://github.com/odoo/odoo/pull/22105#issuecomment-362197988.

feketemihai commented:
@PCatinean Do you have some time to test it ?

jcr-odoo commented:
Do you need to split the base amount from which the tax is calculated? or do you just need to split the tax amount? 

@jco-odoo @rgo-odoo review the PR when they have the time

feketemihai commented:
@jcr-odoo Yes, i need to somehow split the "base" amount, see https://github.com/feketemihai/odoo/blob/13.0_imp_l10n_ro/addons/l10n_ro/data/account_tax_data.xml#L1567,
            (0,0, {
                'factor_percent': 100,
                'repartition_type': 'base',
                'plus_report_line_ids': [ref('account_tax_report_ro_baza_rd22'),
                                         ref('account_tax_report_ro_baza_rd9')],
            }),

i should split as:
 (0,0, {
                'factor_percent': 100,
                'repartition_type': 'base',
                'plus_report_line_ids': [ref('account_tax_report_ro_baza_rd22')],
            }),
            (0,0, {
                'factor_percent': 50,
                'repartition_type': 'base',
                'plus_report_line_ids': [ref('account_tax_report_ro_baza_rd9')],
            }),

feketemihai commented:
@jco-odoo Do you think we can add an extra_base option into repartition_type, and to add a  account move line with 0 values in debit, credit, only to have the tax_base_amount calculated from extra_base line factor and base amount, so we can show it on the tax report?

jco-odoo commented:
@feketemihai Nice hearing from you again.   Indeed, the base amount is always 100% and only oàne line.  You can still use the group of taxes as before, but I would be surprised that this could solve it.  On the other hand, if I understand correctly, this is mainly done for the report.  And there, you could just add a line with a formula.

feketemihai commented:
@jco-odoo Thanks for the answer, i will adapt then the repartition lines in the next days.

Also, in most of Eastern Europe countries use Storno Accounting, not Contra accouting, which is the default behaviour in Odoo, do you think we can include in the next version, i think for Odoo will be good at the end because lots of companies can't use the Base Accounting. If you accept i can try to work on master branch to include it. We are discussing this for a very long time, i think it's almost 9-10 years, but with no actual result in base modules, so all the time we have to drop constraint at account.move.line level, and add a new one, which for a lot of records, it will take a very long time ar updating the modules.

jco-odoo commented:
@feketemihai @qdp-odoo Main reason is just lack of time (and effort to check all the code where it can give problems) and an easy excuse is somehow that most even of those countries using storno are probably not legally required to do so.  That does not mean we should not do it, but if you have some experience on the consequences in the code and some better clarifications on when it is required and how it could convince more people to use Odoo, it can help us forward.

feketemihai commented:
@jco-odoo Done with the changes, i made some small changes since i noticed 2 probably wrong behaviour with this distribution in tax report:
1. when you have on invoice line one tax with no distribution of the base amount (no 'plus_report_line_ids' or 'minus_report_line_ids' set, that's why i added 'plus_report_line_ids': [],
'minus_report_line_ids': [] on the ones that should not be distributed on base), the second line a tax with distribution of base amount, the distribution of the second one is used also for the first - didn't have time to check the code  
2. When you have a Tax on Sale with Cash Basis, the distribution lines should be set inverse, (since the payment normally is on credit, if you set a plus line it will substract the amount from tax report not added), that's why i defined them as:
invoice_repartition_line_ids with  minus_report_line_ids,
refund_repartition_line_ids with  plus_report_line_ids
I already checked in code and it seems that they are taxes in l10n_fr defined with plus_report_line_ids, so i think somebody should check it.

rgo-odoo commented on addons/l10n_ro/data/account_tax_report_data.xml:
```diff
@@ -1,405 +1,901 @@
-        <field name="code">ROTAX_TVA_colectata_5_Baza</field>
+    
+    <record id="account_tax_report_ro_baza_rd3" model="account.tax.report.line">
+        <field name="name">3 - BAZA - Livrări de bunuri sau prestări de servicii pentru care locul livrării/locul prestării este în afara României (în UE sau în afara UE), precum şi livrări intracomunitare de bunuri, scutite conform art. 294 alin.(2) lit.b) şi c) din Codul fiscal, din care:</field>
```
can text for few lines(like this one) be truncated(something similar to what we have done <a href="https://github.com/odoo/odoo/blob/13.0/addons/l10n_lu/data/account_tax_report_line.xml#L295">here</a>) such that their Balance can be seen without scrolling right?
![captured](https://user-images.githubusercontent.com/7601649/69130540-b0942780-0ad6-11ea-9e8a-bf1aee993df8.gif)

feketemihai replied:
I updated and test the report with 10.000.000 amount, no need to scroll anymore.

feketemihai commented on addons/l10n_ro/data/account_tax_group_data.xml:
```diff
@@ -0,0 +1,49 @@
+    <record id="tax_group_tva_19" model="account.tax.group">
+        <field name="name">TVA 19%</field>
+    </record>
+    <record id="tax_group_tva_0_eu" model="account.tax.group">
```
Assigned taxes to the new tax groups

feketemihai commented on addons/l10n_ro/data/account_group_data.xml:
```diff
@@ -0,0 +1,714 @@
+        <field name="name">CLASELE 1-7</field>
+    </record>
+    <record id="account_group_cls_15" model="account.group">
+        <field name="name">CLASELE 1-6</field>
```
Updated typo

feketemihai commented:
@rgo-odoo I updated the comments. Thanks for review.

feketemihai commented:
@jcr-odoo @jco-odoo Can you check it again, please.

feketemihai commented:
@rgo-odoo Any news on this one?

rgo-odoo commented:
@feketemihai It mostly looks good to me. Now @jco-odoo will be doing final review before merging it.

feketemihai commented:
@rgo-odoo Thanks for feedback.

jco-odoo commented on addons/l10n_ro/data/account_tax_report_data.xml:
```diff
@@ -1,405 +1,901 @@
+    
+    <record id="account_tax_report_ro_baza_rd31" model="account.tax.report.line">
+        <field name="name">3.1 - BAZA - Prestări de servicii intracomunitare care nu beneficiază de scutire in statul membru in care taxa este datorată</field>
+        <field name="tag_name">3.1 - BAZA - Prestări de servicii intracomunitare care nu beneficiază de scutire in statul membru in care taxa este datorată</field>
```
It is better to have short tag names as they will be shown in the many2many tags (tax grid) for configuring taxes.  Also, the names itself should not be made very long to avoid to need to scroll horizontally.

feketemihai commented:
@rgo-odoo @jco-odoo I will focus then on master. Thanks. Can you please close all PR's...
https://github.com/odoo/odoo/pulls?utf8=%E2%9C%93&q=is%3Apr+is%3Aopen+l10n_ro

rgo-odoo commented:
@feketemihai We took care of improvements suggested by JCO and created another branch targeting <a href="https://github.com/odoo/odoo/compare/master...odoo-dev:master-l10n_ro-hpr">master</a>. Apart from those improvements, we merged `Regim Intra-Comunitar Bunuri` and `Regim Intra-Comunitar Servicii` fiscal positions to have only one, that is, `Regim Intra-Comunitar`. To achieve the goal, we kept separate taxes for `services/bunuri` and mapped them in `Regim Intra-Comunitar`. Following this way, we could use auto applicability feature for few fiscal positions.
It would be great if you could give it a try here, http://runbot.odoo.com/runbot/repo/git-github-com-odoo-dev-enterprise-8?search=master-l10n_ro-hpr and share your feedback on it.
thanks.

feketemihai commented:
@rgo-odoo I get your point, but will be more complicated to maintain, just for the record we have "Taxare Inversa", "Regim TVA la Incasare" where we should also map this new taxes on this fiscal positions (will check the branch and update if i have write access), also will be difficult for the user because they will have to check at product creation if it's a service, to put the right taxes, since on the chart configuration we only have one default tax. We can do it with default values... I will test it and let you know. Thanks.

feketemihai commented:
@rgo-odoo I created https://github.com/odoo-dev/odoo/pull/174 for tax mapping of services taxes to missing ffiscal positions. I will test it again on runbot after you merge the PR.

rgo-odoo commented:
@feketemihai thanks for the reply and updating other fiscal positions. I merged your changes into #43095, you can now do further testing here, http://runbot.odoo.com/runbot/repo/git-github-com-odoo-dev-odoo-2?search=master-l10n_ro-hpr and share your opinion on the result.

feketemihai commented:
@rgo-odoo  I updated now the tax names, to be more clear that there are  for goods, updated the tax report lines since i checked now and there were two more lines added. Can you please merge it, and also on the enterprise branch, so i can test also the tax report. Thanks.

feketemihai commented:
@rgo-odoo https://github.com/odoo-dev/odoo/pull/175

rgo-odoo commented:
@feketemihai I just integrated your PR into mine and added one commit on top of your commit to fix few conflicting IDs and formulas, see https://github.com/odoo/odoo/pull/43095/commits/e2a49033abbf5850a854f247b3247783dbddf3a5. can you validate it and give it a try on <a href="http://runbot.odoo.com/runbot/repo/git-github-com-odoo-dev-enterprise-8?search=master-l10n_ro-hpr"/>runbot</a> ? thanks.

feketemihai commented:
@rgo-odoo Hi Ravi, i tested locally and adjusted some of them, to be mapped correctly to the Tax Report. https://github.com/odoo-dev/odoo/pull/177 Please merge it.

You can merge in master, and i will continue work after to add the Tags for Balance Sheet and Profit and Loss.

rgo-odoo commented:
@feketemihai thanks for the changes. I have pushed your changes to mine branch.

pedrobaeza commented:
Closing as per comment in #22105
